include/translated-term.php

```diff
@@ -338,4 +338,181 @@ protected function get_db_infos() {
+	 *     @type int      $parent       The id of the parent term. Default 0.
+	 *     @type string   $slug         The term slug to use. Default empty string.
+	 *     @type string[] $translations The translation group to assign to the term with language slug as keys and `term_id` as values.
+	 * }
+	 * @return array|WP_Error {
+	 *     An array of the new term data, `WP_Error` otherwise.
+	 *
+	 *     @type int        $term_id          The new term ID.
+	 *     @type int|string $term_taxonomy_id The new term taxonomy ID. Can be a numeric string.
+	 * }
+	 */
+	public function insert( string $term, string $taxonomy, PLL_Language $language, $args = array() ) {
+		$language_callback = $this->get_slug_management_language_callback( $language );
+		$parent_callback   = $this->get_slug_management_parent_callback( $args['parent'] ?? 0 );
+
+		$this->toggle_slug_management( $language_callback, $parent_callback );
+		$term = wp_insert_term( $term, $taxonomy, $args );
+		$this->toggle_slug_management( $language_callback, $parent_callback );
+
+		if ( is_wp_error( $term ) ) {
+			// Something went wrong!
+			return $term;
+		}
+
+		$this->model->term->set_language( (int) $term['term_id'], $language );
+
+		if ( ! empty( $args['translations'] ) ) {
+			$this->model->term->save_translations(
+				(int) $term['term_id'],
+				array_merge(
+					$args['translations'],
+					array(
+						$language->slug => (int) $term['term_id'],
+					)
+				)
+			);
+		}
+
+		return $term;
+	}
+
+	/**
+	 * Wraps `wp_update_term` with language feature.
+	 *
+	 * @since 3.7
+	 *
+	 * @param int   $term_id The ID of the term.
+	 * @param array $args {
+	 *     Optional. Array of arguments for updating a term.
+	 *
+	 *     @type string       $alias_of     Slug of the term to make this term an alias of.
+	 *                                      Default empty string. Accepts a term slug.
+	 *     @type string       $description  The term description. Default empty string.
+	 *     @type int          $parent       The id of the parent term. Default 0.
+	 *     @type string       $slug         The term slug to use. Default empty string.
+	 *     @type PLL_Language $lang         The term language object.
+	 *     @type string[]     $translations The translation group to assign to the term with language slug as keys and `term_id` as values.
+	 * }
+	 * @return array|WP_Error An array containing the `term_id` and `term_taxonomy_id`,
+	 *                        WP_Error otherwise.                  WP_Error otherwise.
+	 */
+	public function update( int $term_id, array $args = array() ) {
+		$term = get_term( $term_id );
+		if ( ! $term instanceof WP_Term ) {
+			return new WP_Error( 'invalid_term', __( 'Empty Term.', 'polylang' ) );
+		}
+
+		$new_language = ! empty( $args['lang'] ) ? $this->model->get_language( $args['lang'] ) : null;
+		if ( false === $new_language ) {
+			return new WP_Error( 'invalid_language', __( 'Please provide a valid language.', 'polylang' ) );
+		}
```

Comment:
To simplify and improve the readability,I would manage the language in a unique block instead of 2 separate:
```php
	$language = $this->get_language( $term_id );
	if ( ! empty( $args['lang'] ) ) {
		$language = $this->model->get_language( $args['lang'] );
		if ( ! $language instance of PLL_Language ) {
			return new WP_Error( 'invalid_language', __( 'Please provide a valid language.', 'polylang' ) );
		}
		
		$this->set_language( $term_id, $language );
	}
```

Reply:
Done in b54df4000a24ad2807375b7fbb6d623ea1dfe64c.